### DISCUSSION TITLE
Need help in setting up Configuration for Dial Assemble API

### DISCUSSION THREAD

**Discussion by nidhikachhawa24** [Sunbird-Ed/Community#462] [Installation] [answered]
**Scenario:**
While scanning QR code from mobile app, getting error _**Due to technical error, we're unable to process the request.**_

**Analysis:**
Tried to check the assemble API - getting **_404 error_**

**Help:**
I need help in setting up the necessary configurations needed by the dial assemble API.

cc: @vinukumar-vs @rajeevsathish @Krishnaj20 @rajeshkumaravel @swayangjit

**Comment by raghupathiguduri** [comment]
As assemble api related to lms service, we checked the lms service logs. we found below error message in the lms logs. 

`lms-7b6d5588c7-97wcl lms 2023-04-13T20:26:19.657036830+05:30 {"timestamp":"2023-04-13T14:56:19.656Z","msg":"Requested page does not exist.","lname":"defaultLogger","tname":"Thread-40475","level":"ERROR","stack_trace":"<#56e34bff> org.sunbird.common.exception.ProjectCommonException: Requested page does not exist.\n\tat org.sunbird.learner.actors.PageManagementActor.getDIALPageData(PageManagementActor.java:853)\n\tat org.sunbird.learner.actors.PageManagementActor.onReceive(PageManagementActor.java:87)\n\tat org.sunbird.actor.base.BaseActor.onReceive(BaseActor.java:23)\n\tat akka.actor.UntypedAbstractActor$$anonfun$receive$1.applyOrElse(AbstractActor.scala:337)\n\tat akka.actor.Actor$class.aroundReceive(Actor.scala:539)\n\tat akka.actor.AbstractActor.aroundReceive(AbstractActor.scala:226)\n\tat akka.actor.ActorCell.receiveMessage(ActorCell.scala:610)\n\tat akka.actor.ActorCell.invoke(ActorCell.scala:579)\n\tat akka.dispatch.Mailbox.processMailbox(Mailbox.scala:268)\n\tat akka.dispatch.Mailbox.run(Mailbox.scala:229)\n\tat akka.dispatch.Mailbox.exec(Mailbox.scala:241)\n\tat akka.dispatch.forkjoin.ForkJoinTask.doExec(ForkJoinTask.java:260)\n\tat akka.dispatch.forkjoin.ForkJoinPool$WorkQueue.runTask(ForkJoinPool.java:1339)\n\tat akka.dispatch.forkjoin.ForkJoinPool.runWorker(ForkJoinPool.java:1979)\n\tat akka.dispatch.forkjoin.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:107)\n","HOSTNAME":"lms-7b6d5588c7-97wcl","application.home":"/home/sunbird/lms/lms-service-1.0-SNAPSHOT"}
lms-7b6d5588c7-97wcl lms 2023-04-13T20:26:19.657056625+05:30 2023-04-13 14:56:19,656 Requested page does not exist.
lms-7b6d5588c7-97wcl lms 2023-04-13T20:26:19.657059299+05:30 org.sunbird.common.exception.ProjectCommonException: Requested page does not exist.
lms-7b6d5588c7-97wcl lms 2023-04-13T20:26:19.657061284+05:30 	at org.sunbird.learner.actors.PageManagementActor.getDIALPageData(PageManagementActor.java:853)
lms-7b6d5588c7-97wcl lms 2023-04-13T20:26:19.657063382+05:30 	at org.sunbird.learner.actors.PageManagementActor.onReceive(PageManagementActor.java:87)
lms-7b6d5588c7-97wcl lms 2023-04-13T20:26:19.657065374+05:30 	at org.sunbird.actor.base.BaseActor.onReceive(BaseActor.java:23)
lms-7b6d5588c7-97wcl lms 2023-04-13T20:26:19.657077801+05:30 	at akka.actor.UntypedAbstractActor$$anonfun$receive$1.applyOrElse(AbstractActor.scala:337)
lms-7b6d5588c7-97wcl lms 2023-04-13T20:26:19.657080630+05:30 	at akka.actor.Actor$class.aroundReceive(Actor.scala:539)
lms-7b6d5588c7-97wcl lms 2023-04-13T20:26:19.657083622+05:30 	at akka.actor.AbstractActor.aroundReceive(AbstractActor.scala:226)
lms-7b6d5588c7-97wcl lms 2023-04-13T20:26:19.657086705+05:30 	at akka.actor.ActorCell.receiveMessage(ActorCell.scala:610)
lms-7b6d5588c7-97wcl lms 2023-04-13T20:26:19.657090486+05:30 	at akka.actor.ActorCell.invoke(ActorCell.scala:579)
lms-7b6d5588c7-97wcl lms 2023-04-13T20:26:19.657093338+05:30 	at akka.dispatch.Mailbox.processMailbox(Mailbox.scala:268)
lms-7b6d5588c7-97wcl lms 2023-04-13T20:26:19.657096167+05:30 	at akka.dispatch.Mailbox.run(Mailbox.scala:229)
lms-7b6d5588c7-97wcl lms 2023-04-13T20:26:19.657098752+05:30 	at akka.dispatch.Mailbox.exec(Mailbox.scala:241)
lms-7b6d5588c7-97wcl lms 2023-04-13T20:26:19.657101496+05:30 	at akka.dispatch.forkjoin.ForkJoinTask.doExec(ForkJoinTask.java:260)
lms-7b6d5588c7-97wcl lms 2023-04-13T20:26:19.657104229+05:30 	at akka.dispatch.forkjoin.ForkJoinPool$WorkQueue.runTask(ForkJoinPool.java:1339)
lms-7b6d5588c7-97wcl lms 2023-04-13T20:26:19.657107016+05:30 	at akka.dispatch.forkjoin.ForkJoinPool.runWorker(ForkJoinPool.java:1979)
lms-7b6d5588c7-97wcl lms 2023-04-13T20:26:19.657109754+05:30 	at akka.dispatch.forkjoin.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:107)
lms-7b6d5588c7-97wcl lms 2023-04-13T20:26:19.657242347+05:30 2023-04-13 14:56:19,657 {"eid":"ERROR","ets":1681397779657,"ver":"3.0","mid":"6edd4f98-02c9-3acd-125d-a028977825b1","actor":{"id":"internal","type":"Consumer"},"context":{"channel":"0124487522476933120","pdata":{"id":"production.production.portal","pid":"lms-service","ver":"1.0"},"env":"Miscellaneous","did":"6f78c19060101e1106dd020e653be6e7","cdata":[{"id":"6edd4f98-02c9-3acd-125d-a028977825b1","type":"Request"}],"rollup":{}},"edata":{"err":"PAGE_NOT_EXIST","stacktrace":"org.sunbird.learner.actors.PageManagementActor.getDIALPageData(PageManagementActor.java:853)org.sunb","errtype":"api_access"}}
lms-7b6d5588c7-97wcl lms 2023-04-13T20:26:19.657476315+05:30 2023-04-13 14:56:19,657 {"eid":"ERROR","ets":1681397779657,"ver":"3.0","mid":"6edd4f98-02c9-3acd-125d-a028977825b1","actor":{"id":"internal","type":"Consumer"},"context":{"channel":"0124487522476933120","pdata":{"id":"production.production.portal","pid":"lms-service","ver":"1.0"},"env":"Miscellaneous","did":"6f78c19060101e1106dd020e653be6e7","cdata":[{"id":"6edd4f98-02c9-3acd-125d-a028977825b1","type":"Request"}],"rollup":{}},"edata":{"err":"404","stacktrace":"org.sunbird.learner.actors.PageManagementActor.getDIALPageData(PageManagementActor.java:853)\norg.sun","errtype":"PAGE_NOT_EXIST"}}
lms-7b6d5588c7-97wcl lms 2023-04-13T20:26:19.657495147+05:30 {"timestamp":"2023-04-13T14:56:19.657Z","msg":"Requested page does not exist.","lname":"defaultLogger","tname":"Thread-40475","level":"ERROR","stack_trace":"<#56e34bff> org.sunbird.common.exception.ProjectCommonException: Requested page does not exist.\n\tat org.sunbird.learner.actors.PageManagementActor.getDIALPageData(PageManagementActor.java:853)\n\tat org.sunbird.learner.actors.PageManagementActor.onReceive(PageManagementActor.java:87)\n\tat org.sunbird.actor.base.BaseActor.onReceive(BaseActor.java:23)\n\tat akka.actor.UntypedAbstractActor$$anonfun$receive$1.applyOrElse(AbstractActor.scala:337)\n\tat akka.actor.Actor$class.aroundReceive(Actor.scala:539)\n\tat akka.actor.AbstractActor.aroundReceive(AbstractActor.scala:226)\n\tat akka.actor.ActorCell.receiveMessage(ActorCell.scala:610)\n\tat akka.actor.ActorCell.invoke(ActorCell.scala:579)\n\tat akka.dispatch.Mailbox.processMailbox(Mailbox.scala:268)\n\tat akka.dispatch.Mailbox.run(Mailbox.scala:229)\n\tat akka.dispatch.Mailbox.exec(Mailbox.scala:241)\n\tat akka.dispatch.forkjoin.ForkJoinTask.doExec(ForkJoinTask.java:260)\n\tat akka.dispatch.forkjoin.ForkJoinPool$WorkQueue.runTask(ForkJoinPool.java:1339)\n\tat akka.dispatch.forkjoin.ForkJoinPool.runWorker(ForkJoinPool.java:1979)\n\tat akka.dispatch.forkjoin.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:107)\n","HOSTNAME":"lms-7b6d5588c7-97wcl","application.home":"/home/sunbird/lms/lms-service-1.0-SNAPSHOT"}
lms-7b6d5588c7-97wcl lms 2023-04-13T20:26:19.657557789+05:30 2023-04-13 14:56:19,657 Requested page does not exist.
lms-7b6d5588c7-97wcl lms 2023-04-13T20:26:19.657567699+05:30 org.sunbird.common.exception.ProjectCommonException: Requested page does not exist.
lms-7b6d5588c7-97wcl lms 2023-04-13T20:26:19.657570765+05:30 	at org.sunbird.learner.actors.PageManagementActor.getDIALPageData(PageManagementActor.java:853)
lms-7b6d5588c7-97wcl lms 2023-04-13T20:26:19.657572737+05:30 	at org.sunbird.learner.actors.PageManagementActor.onReceive(PageManagementActor.java:87)
lms-7b6d5588c7-97wcl lms 2023-04-13T20:26:19.657574800+05:30 	at org.sunbird.actor.base.BaseActor.onReceive(BaseActor.java:23)
lms-7b6d5588c7-97wcl lms 2023-04-13T20:26:19.657576720+05:30 	at akka.actor.UntypedAbstractActor$$anonfun$receive$1.applyOrElse(AbstractActor.scala:337)
lms-7b6d5588c7-97wcl lms 2023-04-13T20:26:19.657578573+05:30 	at akka.actor.Actor$class.aroundReceive(Actor.scala:539)
lms-7b6d5588c7-97wcl lms 2023-04-13T20:26:19.657580463+05:30 	at akka.actor.AbstractActor.aroundReceive(AbstractActor.scala:226)
lms-7b6d5588c7-97wcl lms 2023-04-13T20:26:19.657582691+05:30 	at akka.actor.ActorCell.receiveMessage(ActorCell.scala:610)
lms-7b6d5588c7-97wcl lms 2023-04-13T20:26:19.657584583+05:30 	at akka.actor.ActorCell.invoke(ActorCell.scala:579)
lms-7b6d5588c7-97wcl lms 2023-04-13T20:26:19.657586497+05:30 	at akka.dispatch.Mailbox.processMailbox(Mailbox.scala:268)
lms-7b6d5588c7-97wcl lms 2023-04-13T20:26:19.657588606+05:30 	at akka.dispatch.Mailbox.run(Mailbox.scala:229)
lms-7b6d5588c7-97wcl lms 2023-04-13T20:26:19.657590632+05:30 	at akka.dispatch.Mailbox.exec(Mailbox.scala:241)
lms-7b6d5588c7-97wcl lms 2023-04-13T20:26:19.657592454+05:30 	at akka.dispatch.forkjoin.ForkJoinTask.doExec(ForkJoinTask.java:260)
lms-7b6d5588c7-97wcl lms 2023-04-13T20:26:19.657594292+05:30 	at akka.dispatch.forkjoin.ForkJoinPool$WorkQueue.runTask(ForkJoinPool.java:1339)
lms-7b6d5588c7-97wcl lms 2023-04-13T20:26:19.657596139+05:30 	at akka.dispatch.forkjoin.ForkJoinPool.runWorker(ForkJoinPool.java:1979)
lms-7b6d5588c7-97wcl lms 2023-04-13T20:26:19.657598023+05:30 	at akka.dispatch.forkjoin.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:107)
lms-7b6d5588c7-97wcl lms 2023-04-13T20:26:19.657702960+05:30 2023-04-13 14:56:19,657 {"eid":"ERROR","ets":1681397779657,"ver":"3.0","mid":"6edd4f98-02c9-3acd-125d-a028977825b1","actor":{"id":"internal","type":"Consumer"},"context":{"channel":"0124487522476933120","pdata":{"id":"production.production.portal","pid":"lms-service","ver":"1.0"},"env":"Miscellaneous","did":"6f78c19060101e1106dd020e653be6e7","cdata":[{"id":"6edd4f98-02c9-3acd-125d-a028977825b1","type":"Request"}],"rollup":{}},"edata":{"err":"PAGE_NOT_EXIST","stacktrace":"org.sunbird.learner.actors.PageManagementActor.getDIALPageData(PageManagementActor.java:853)org.sunb","errtype":"api_access"}}
lms-7b6d5588c7-97wcl lms 2023-04-13T20:26:34.808353447+05:30 {}
lms-7b6d5588c7-97wcl lms 2023-04-13T20:26:34.808434962+05:30 2023-04-13 14:56:34,808 {"actor":{"id":"f64268b3-2fcc-4fac-aefa-a11bf4b0cad2","type":"user"},"eid":"LOG","edata":{"level":"INFO","requestid":"0a841f98-fcb0-48a0-d484-058ac671ed0b","type":"system","message":"CourseEnrolmentActor :: list :: UserId = f64268b3-2fcc-4fac-aefa-a11bf4b0cad2"},"ver":"3.0","ets":1681397794808,"context":{"channel":"course-service","pdata":{"ver":"1.0","pid":"course-service","id":"org.sunbird.course-service"},"env":"","did":"6f04fab58d1e4d1b25dedd387bce5f37","sid":"qsrxx4J4ok0fxvDxnaCPEvAUWe40N8Uf"},"mid":"LOG:69a90ba9-d4ea-49c4-8c43-c517cb05ea53"}`

It seems we need to create some page assemble for this to work. But we don't have details on how to create the page assemble and what we need to configure on that.

**Reply by raghupathiguduri** [reply]
We somehow managed to add data to the page_management and page_section table with the help of other contributor of sunbird. But now we are seeing null pointer exception in the logs of lms service.

`lms-58b5dbcf99-lzts7 lms 2023-04-14T12:42:03.257331321+05:30 {"timestamp":"2023-04-14T07:12:03.257Z","msg":"Fetching data from Cache for NA:DIAL Code Consumption","lname":"defaultLogger","tname":"application-akka.actor.default-dispatcher-10","level":"INFO","HOSTNAME":"lms-58b5dbcf99-lzts7","application.home":"/home/sunbird/lms/lms-service-1.0-SNAPSHOT"}
lms-58b5dbcf99-lzts7 lms 2023-04-14T12:42:03.257364396+05:30 2023-04-14 07:12:03,257 Fetching data from Cache for NA:DIAL Code Consumption
lms-58b5dbcf99-lzts7 lms 2023-04-14T12:42:03.257525674+05:30 {"timestamp":"2023-04-14T07:12:03.257Z","msg":"PageManagementActor:getPageData: Exception occurred with error message = null","lname":"defaultLogger","tname":"application-akka.actor.default-dispatcher-10","level":"ERROR","stack_trace":"<#2f71e36d> java.lang.NullPointerException: null\n\tat org.sunbird.learner.actors.PageManagementActor.getContentData(PageManagementActor.java:662)\n\tat org.sunbird.learner.actors.PageManagementActor.getSectionData(PageManagementActor.java:367)\n\tat org.sunbird.learner.actors.PageManagementActor.getDIALPageData(PageManagementActor.java:867)\n\tat org.sunbird.learner.actors.PageManagementActor.onReceive(PageManagementActor.java:87)\n\tat org.sunbird.actor.base.BaseActor.onReceive(BaseActor.java:23)\n\tat akka.actor.UntypedAbstractActor$$anonfun$receive$1.applyOrElse(AbstractActor.scala:337)\n\tat akka.actor.Actor$class.aroundReceive(Actor.scala:539)\n\tat akka.actor.AbstractActor.aroundReceive(AbstractActor.scala:226)\n\tat akka.actor.ActorCell.receiveMessage(ActorCell.scala:610)\n\tat akka.actor.ActorCell.invoke(ActorCell.scala:579)\n\tat akka.dispatch.Mailbox.processMailbox(Mailbox.scala:268)\n\tat akka.dispatch.Mailbox.run(Mailbox.scala:229)\n\tat akka.dispatch.Mailbox.exec(Mailbox.scala:241)\n\tat akka.dispatch.forkjoin.ForkJoinTask.doExec(ForkJoinTask.java:260)\n\tat akka.dispatch.forkjoin.ForkJoinPool$WorkQueue.runTask(ForkJoinPool.java:1339)\n\tat akka.dispatch.forkjoin.ForkJoinPool.runWorker(ForkJoinPool.java:1979)\n\tat akka.dispatch.forkjoin.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:107)\n","HOSTNAME":"lms-58b5dbcf99-lzts7","application.home":"/home/sunbird/lms/lms-service-1.0-SNAPSHOT"}
lms-58b5dbcf99-lzts7 lms 2023-04-14T12:42:03.257789079+05:30 2023-04-14 07:12:03,257 PageManagementActor:getPageData: Exception occurred with error message = null
lms-58b5dbcf99-lzts7 lms 2023-04-14T12:42:03.257815909+05:30 java.lang.NullPointerException: null
lms-58b5dbcf99-lzts7 lms 2023-04-14T12:42:03.257819503+05:30 	at org.sunbird.learner.actors.PageManagementActor.getContentData(PageManagementActor.java:662)
lms-58b5dbcf99-lzts7 lms 2023-04-14T12:42:03.257822144+05:30 	at org.sunbird.learner.actors.PageManagementActor.getSectionData(PageManagementActor.java:367)
lms-58b5dbcf99-lzts7 lms 2023-04-14T12:42:03.257824704+05:30 	at org.sunbird.learner.actors.PageManagementActor.getDIALPageData(PageManagementActor.java:867)
lms-58b5dbcf99-lzts7 lms 2023-04-14T12:42:03.257828146+05:30 	at org.sunbird.learner.actors.PageManagementActor.onReceive(PageManagementActor.java:87)
lms-58b5dbcf99-lzts7 lms 2023-04-14T12:42:03.257831302+05:30 	at org.sunbird.actor.base.BaseActor.onReceive(BaseActor.java:23)
lms-58b5dbcf99-lzts7 lms 2023-04-14T12:42:03.257834153+05:30 	at akka.actor.UntypedAbstractActor$$anonfun$receive$1.applyOrElse(AbstractActor.scala:337)
lms-58b5dbcf99-lzts7 lms 2023-04-14T12:42:03.257837039+05:30 	at akka.actor.Actor$class.aroundReceive(Actor.scala:539)
lms-58b5dbcf99-lzts7 lms 2023-04-14T12:42:03.257841216+05:30 	at akka.actor.AbstractActor.aroundReceive(AbstractActor.scala:226)
lms-58b5dbcf99-lzts7 lms 2023-04-14T12:42:03.257844190+05:30 	at akka.actor.ActorCell.receiveMessage(ActorCell.scala:610)
lms-58b5dbcf99-lzts7 lms 2023-04-14T12:42:03.257847076+05:30 	at akka.actor.ActorCell.invoke(ActorCell.scala:579)
lms-58b5dbcf99-lzts7 lms 2023-04-14T12:42:03.257849912+05:30 	at akka.dispatch.Mailbox.processMailbox(Mailbox.scala:268)
lms-58b5dbcf99-lzts7 lms 2023-04-14T12:42:03.257852744+05:30 	at akka.dispatch.Mailbox.run(Mailbox.scala:229)
lms-58b5dbcf99-lzts7 lms 2023-04-14T12:42:03.257855444+05:30 	at akka.dispatch.Mailbox.exec(Mailbox.scala:241)
lms-58b5dbcf99-lzts7 lms 2023-04-14T12:42:03.257858488+05:30 	at akka.dispatch.forkjoin.ForkJoinTask.doExec(ForkJoinTask.java:260)
lms-58b5dbcf99-lzts7 lms 2023-04-14T12:42:03.257861446+05:30 	at akka.dispatch.forkjoin.ForkJoinPool$WorkQueue.runTask(ForkJoinPool.java:1339)
lms-58b5dbcf99-lzts7 lms 2023-04-14T12:42:03.257864241+05:30 	at akka.dispatch.forkjoin.ForkJoinPool.runWorker(ForkJoinPool.java:1979)
lms-58b5dbcf99-lzts7 lms 2023-04-14T12:42:03.257867011+05:30 	at akka.dispatch.forkjoin.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:107)`

Data that we added into cassandra tables.

`cqlsh> SELECT * from sunbird.page_management WHERE id='01269259730241945613';

 id                   | appmap                                              | created_date | createdby                            | createddate                  | name                  | organisationid | portalmap                                           | updated_date | updatedby                           | updateddate
----------------------+-----------------------------------------------------+--------------+--------------------------------------+------------------------------+-----------------------+----------------+-----------------------------------------------------+--------------+-------------------------------------+------------------------------
 01269259730241945613 | [{"id":"01269259447543398412","index":1,"group":1}] |         null | null | 2019-02-05 17:51:15:071+0000 | DIAL Code Consumption |             NA | [{"id":"01269259447543398412","index":1,"group":1}] |         null | null | 2020-05-11 13:44:29:378+0000

(1 rows)

cqlsh> SELECT * from sunbird.page_section where id='01269259447543398412';

 id                   | alt  | created_date | createdby | createddate | datasource | description | display                          | dynamicfilters | imgurl | name           | searchquery                                                                                                                                                                                                                                                                                                                                                                                                                                                         | sectiondatatype | status | updated_date | updatedby | updateddate
----------------------+------+--------------+-----------+-------------+------------+-------------+----------------------------------+----------------+--------+----------------+---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+-----------------+--------+--------------+-----------+-------------
 01269259447543398412 | null |         null |      null |        null |       null |        null | {"name":{"en":"Linked Content"}} |           null |   null | Linked Content | {"request":{"facets":["language","grade","domain","contentType","primaryCategory","subject","medium"],"filters":{"primaryCategory":["Digital Textbook","Textbook Unit","Course","Explanation Content","Content Playlist"],"contentType":["Collection","TextBook","TextBookUnit","Resource","Course"],"status":["Live"],"compatibilityLevel":{"max":4,"min":1},"visibility":[]},"mode":"collection","userProfile":{}},"limit":10,"sort_by":{"lastUpdatedOn":"desc"}} |         content |      1 |         null |      null |        null

(1 rows)`

**Reply by swayangjit** [reply, marked as answer]
@raghupathiguduri  https://lern.sunbird.org/learn/product-and-developer-guide/batch-service/api-documentation#data-v1-page-create
This is the  documentation of APIs
